{{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/users/websocket.py

```diff
@@ -1,13 +1,18 @@
 async def websocket_application(scope, receive, send):
```

Comment:
I don't understand why we need to move this file. Is it solely for having a place to put the tests?

I think I'm missing how it would grow in a full application... In a WSGI app, the `wsgi.py` isn't usually tested, and act as a project-level file, it doesn't belong to any Django app. I thought this `websocket.py` would also be falling into such category, but by moving it in the users app, I expect it to be specialised to the users-related functionality, but it's not the case yet...

Reply:
It's moved so that you can access models, like the User model. In my latest project, from ASGI, I created a URLRouter that would allow multiple "websocket_application" with some extra security features (CSRF + Session). `websocket.py` was meant to handle all the application code, but my own project started to fill up with multiple different "handle" methods on every message sent through the socket to the server.

The ASGI file points towards the websocket application if the scheme is `wss`.